### ISSUE TITLE
Convert Loudspeaker to mic 

### ISSUE THREAD

**Issue by Hamza123Imran** [twilio/video-quickstart-swift#307]
Hi, i'm using twilio video call in my iOS app. When i call it runs properly. I want to ask that is there any option that twilio provide in which we can convert or change the option to speak . Like mute and unmute. Can we speak as loudspeaker or in mic?

**Comment by paynerc** [comment]
Good morning @Hamza123Imran,

I'm not sure that I am following your question. Are you asking if you can route the audio output between the receiver speaker and the speakers?

If that is your question, the answer is yes. You can chose the route for audio output. Take a look at our "Configuring audio, video input and output devices" guide, specifically the [Selecting specific Audio routes](https://www.twilio.com/docs/video/ios-v2-configuring-audio-video-inputs-and-outputs#selecting-specific-audio-routes) section. This contains the information you need to change the audio output route between the speaker and the receiver.

Let me know if you need any further help,

Ryan

**Comment by Hamza123Imran** [comment]
Sorry if didn't get it, i'm trying to give authority to the user that he/she can on or off the speaker while a call.

**Comment by paynerc** [comment]
So you are saying that you want to give the user the ability to mute their audio during the call?

You can toggle the `enabled` state of the `TVILocalAudioTrack`. See the [API Docs](https://twilio.github.io/twilio-video-ios/docs/latest/Classes/TVILocalAudioTrack.html#//api/name/enabled) for further information.

**Comment by Hamza123Imran** [comment]
No, i'm not saying for mute, i'm saying that when video call runs and we talk to any other user , the voice goes through speaker, i want that if user wants to talk through microphone how this could be done?

**Comment by paynerc** [comment]
OK, so then my first response still stands. Take a look at our "Configuring audio, video input and output devices" guide, specifically the [Selecting specific Audio routes](
https://www.twilio.com/docs/video/ios-v2-configuring-audio-video-inputs-and-outputs#selecting-specific-audio-routes) section. This contains the information you need to change the audio input and output routes.

Ryan

**Comment by Hamza123Imran** [comment]
I have study the link that you have provided, i have implemented that on my button action like this,
` // Change the audio route after connecting to a Room.
        room.audioDevice.block = {
            do {
                kDefaultAVAudioSessionConfigurationBlock()
                
                let audioSession = AVAudioSession.sharedInstance()
                try audioSession.setMode(AVAudioSessionModeVoiceChat)
            } catch let error as NSError {
                print("Fail: \(error.localizedDescription)")
            }
        }
        
        room.audioDevice.block();`

But i'm getting this error `Value of type 'TVIRoom?' has no member 'audioDevice'`

**Comment by paynerc** [comment]
Sadly, you are correct. The documentation is incorrect. The `audioDevice` is accessible from `TwilioVideo`, not `TVIRoom`. See [`TwilioVideo.audioDevice`](https://twilio.github.io/twilio-video-ios/docs/latest/Classes/TwilioVideo.html#//api/name/audioDevice) in the API docs.

I will create a ticket to correct the documentation I provided earlier.

**Comment by paynerc** [comment]
Just wanted to let you know that we have fixed the issues in the documentation.

Have you had any success with the approach suggested?

**Comment by Hamza123Imran** [comment]
No, still i haven't find any solution for my problem , if you have covered that please guide me for this. @paynerc

**Comment by paynerc** [comment]
@Hamza123Imran,

This sample code block, albeit it in Obj-C, is pretty much what you need to do to switch between the receiver and the speaker.

```
- (IBAction)swapAudioPressed:(id)sender {
    // Get the audioDevice in use by the SDK
    TVIDefaultAudioDevice *audioDevice = TwilioVideo.audioDevice;

    if (self.isSpeakerInUse) {
        // This block will first configure the audio device for the default behavior and then will switch the mode to VoiceChat, which will cause the audio output to be routed to the the receiver
        audioDevice.block = ^{
            kDefaultAVAudioSessionConfigurationBlock();

            NSError *error = nil;
            if (![[AVAudioSession sharedInstance] setMode:AVAudioSessionModeVoiceChat error:&error]) {
                NSLog(@"AVAudioSession setMode:error: %@",error);
            }
        };

        self.speakerInUse = NO;
    } else {
        // This block will revert the audio device to the default behavior, which is routing the audio through the speakers
        audioDevice.block = ^{
            kDefaultAVAudioSessionConfigurationBlock();
        };

        self.speakerInUse = YES;
    }

    audioDevice.block();
}
```

A simple boolean property can then be added to your class to maintain the toggle state of the output.

Let me know if this helps.

Ryan

**Comment by Shobhit-InstaaCoder** [comment]
> @Hamza123Imran,
> 
> It's been a while since I sent you the sample code. Did that work out for you? I am going to close out this issue, but if you have further issues, feel free to reach out.
> 
> Thanks!
> 
> Ryan

Hi When I make video call then receiver end call comes over speaker but who initiate the call they didn't get on speaker mode. I want that call must be on speaker mode at both end. Plz help me out this as I stuck since 1 week.
